test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -61,4 +61,89 @@ public void FirstExperienceWithML()
             });
         }
     }
+
+    public class GithubClassification
+    {
+        public void ClassifyGithubIssues()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+
+            string dataPath = "corefx-issues-train.tsv";
+
+            // Create reader with specific schema. 
+            // string :ID, string: Area, string:Title, string:Description
+            var reader = TextLoader.CreateReader(env, ctx =>
+                                          (area: ctx.LoadText(1),
+                                            title: ctx.LoadText(2),
+                                            description: ctx.LoadText(3),
+                                            dataSource,
+                                            useHeader: true));
+
+            var data = reader.Read(dataPath).AsDynamic;
+
+            var schema = textData.Schema;
+
+            // Verify that the columns are there. 
+            // There ought to be at least one column corresponding to the identifiers in the tuple.
+            CheckSchemaHasColumn(schema, "area", out int areaIdx);
+            CheckSchemaHasColumn(schema, "title", out int titleIdx);
+            CheckSchemaHasColumn(schema, "description", out int descriptionIdx);
+
+            // Next verify they have the expected types.
+            Assert.Equal(TextType.Instance, schema.GetColumnType(areaIdx));
+            Assert.Equal(TextType.Instance, schema.GetColumnType(titleIdx));
+            Assert.Equal(TextType.Instance, schema.GetColumnType(descriptionIdx));
+
+            // sefilipi: Do we need to inspect the data, as in Tom's example?	
+
+            var estimator = Estimator.MakeNew(data)
+                .Append(row => (
+                    // Convert string label to key. 
+                    label: row.area.Dictionarize(),
+                    // featurizes 'description'
+                    description1: row.description.FeaturizeText(),
+                    // featurizes 'title'
+                    title1: row.title.FeaturizeText()))
+                .Append(row => (
+                    // Concatenate the two features into a vector.
+                    // Do we need to make other use of the TextTransform output columns?
+                    features: row.description1.ConcatWith(r.title1),
+                    // preserve the label
+                    label: row.label))
+                // how do we specify arguments for the trainer?
+                .Append(row => r.label.TrainLinearClassification(row.Features));
+
+            var model = estimator.Fit(data);
+
+            string modelPath = "github-Model.zip";
+
+            // we don't currently have the WriteAsync
+            await model.WriteAsync(modelPath);
+        }
+
+        public void PredictGithubIssues()
+        {
+            ClassifyGithubIssues();
+
+            // we don't currently have the ReadAsync
+            var model = await PredictionModel.ReadAsync(ModelPath);
```

Comment:
>ReadAsync [](start = 46, length = 9)

are we planning to keep the typed version of ReadAsync we have for the LearningPipeline?

Reply:
hmm, not sure. .NET folks seem to have a strong opinion on these, @eerhardt ?

---
In reply to: [215088069](https://github.com/dotnet/machinelearning/pull/812#discussion_r215088069) [](ancestors = 215088069)

Reply:
Do we even have a `PredictionModel` class in the new API?

I think the "typed version" comes in the next step, right?  `model.MakePredictionFunction<IssueInput, IssuePrediction>()`.

In general, I think the Read and Write methods should be mirrors of each other. The thing that I "Write" is the thing I should get back when I "Read".

---
In reply to: [215093756](https://github.com/dotnet/machinelearning/pull/812#discussion_r215093756) [](ancestors = 215093756,215088069)

Reply:
Right. I think we used to call this `TransformerChain.LoadFrom`, and the save method is `model.SaveTo`

---
In reply to: [215319607](https://github.com/dotnet/machinelearning/pull/812#discussion_r215319607) [](ancestors = 215319607,215093756,215088069)